### PR TITLE
Bugfix/basepath

### PR DESCRIPTION
**Pull request by Ruwann**
Working towards #1709 

I think we're almost there, some tests I did are now working properly.

Would love to get some feedback/ideas on the implementation and the tests :)

### CODE REVIEW

File: connexion/middleware/main.py [hunk 1]
```diff
@@ -308,6 +308,9 @@ def _build_middleware_stack(self) -> t.Tuple[ASGIApp, t.Iterable[ASGIApp]]:
                 app = middleware(app)  # type: ignore
             apps.append(app)
 
+        # We sort the APIs by base path so that the most specific APIs are registered first.
```
Comment (Ruwann) [inline connexion/middleware/main.py:311]: There might also be an issue with the ordering of routes inside an API (when adding the operations of a spec to an API)

Comment (RobbeSneyders) [reply connexion/middleware/main.py:311]: Indeed, we can probably quite easily reuse the same sorting function there?

File: connexion/utils.py [hunk 2]
```diff
@@ -423,3 +427,44 @@ def inspect_function_arguments(function: t.Callable) -> t.Tuple[t.List[str], boo
...
+
+    class SortableRoute:
+        def __init__(self, path: str) -> None:
+            self.path = path
+            self.path_regex, _, _ = compile_path(path)
+
+        def __lt__(self, other: "SortableRoute") -> bool:
+            return bool(other.path_regex.match(self.path))
+
+    return sorted(routes, key=lambda r: SortableRoute(key(r) if key else r))
```
Comment (RobbeSneyders) [inline connexion/utils.py:460]: Smart :exploding_head:

**Review by RobbeSneyders — COMMENTED** [review]
Thanks @Ruwann!

This approach looks good to me. Unless I'm missing something, I don't think we're sorting paths without path parameters correctly though?

Eg. if you look at the example in the issue, we should also sort `"/swagger"` before `"/"`. I think this is only necessary for the APIs, not for individual routes, but I don't think it has any downside there.

File: connexion/middleware/main.py (hunk 1 above)
Comment (Ruwann) [reply connexion/middleware/main.py:311]: Yes, I'll investigate further in a separate issue as I did a small test that seemed to work fine even though the routes were in the "wrong" order.

**Comment by Ruwann** [comment]
Thanks, good catch. Indeed, this was not yet working properly because the tests added the `/{path:path}` part that's actually needed. Fixed this now :)
Agreed, don't think it has downsides for routes as `/` and `/basepath` are technically not comparable, neither is more specific than the other, but this changes for basepaths.

**Review by RobbeSneyders — APPROVED** [review]
Thanks @Ruwann!

**Pull request merged**